app/controllers/products_controller.rb

```diff
@@ -2,6 +2,7 @@ class ProductsController < ApplicationController
   before_action :find_product, only: %i[show edit update destroy]
 
   def index
+    @paginate = Product.paginate(page: params[:page], per_page:10)
```

Comment:
first you search and then you paginate, it is no use of search if you're paginating first and searching late since now if the page size is like 10 only the search is being applied to first 10 products.

Reply:
also 10 should be a constant in the model